modules/sage/1.0/sage.smk

```diff
@@ -0,0 +1,255 @@
+    output:
+        vcf = temp(CFG["dirs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}--{normal_id}--{pair_status}.vcf"),
+        vcf_gz = CFG["dirs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}--{normal_id}--{pair_status}.vcf.gz"
+    log:
+        stdout = CFG["logs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run_sage.stdout.log",
+        stderr = CFG["logs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run_sage.stderr.log"
+    params:
+        opts = CFG["options"]["sage_run"],
+        chromosomes = get_chromosomes,
+        assembly = lambda w: "hg38" if "38" in str({w.genome_build}) else "hg19",
+        sage= "$(dirname $(readlink -e $(which SAGE)))/sage.jar",
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8)
+    conda:
+        CFG["conda_envs"]["sage"]
+    threads:
+        CFG["threads"]["sage_run"]
+    resources:
+        **CFG["resources"]["sage_run"]
+    shell:
+        op.as_one_line("""
+        echo "running {rule} for {wildcards.tumour_id}--{wildcards.normal_id} on $(hostname)" > {log.stdout}
+        &&
+        java -Xms1G -Xmx{params.jvmheap}m
+        -cp {params.sage} com.hartwig.hmftools.sage.SageApplication
+        -threads {threads}
+        {params.opts}
+        -chr {params.chromosomes}
+        -reference {wildcards.normal_id}
+        -reference_bam {input.normal_bam}
+        -tumor {wildcards.tumour_id} 
+        -tumor_bam {input.tumour_bam}
+        -assembly {params.assembly}
+        -ref_genome {input.fasta}
+        -hotspots {input.hotspots}
+        -panel_bed {input.panel_bed}
+        -high_confidence_bed {input.high_conf_bed}
+        -out {output.vcf}
+        >>  {log.stdout} 2> {log.stderr}
+        && bgzip -c {output.vcf} > {output.vcf_gz}
+        """)
+
+# Filter resulting VCF file on PASS variants
+rule _sage_filter_vcf:
+    input:
+        vcf = str(rules._run_sage.output.vcf_gz)
+    output:
+        vcf_passed = CFG["dirs"]["vcf"] + "combined/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/passed_combined.vcf.gz",
+        vcf_passed_tbi = CFG["dirs"]["vcf"] + "combined/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/passed_combined.vcf.gz.tbi"
+    log:
+        stdout = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/filter_passed.stdout.log",
+        stderr = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/filter_passed.stderr.log"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    threads:
+        CFG["threads"]["filter"]
+    resources:
+        **CFG["resources"]["filter"]
+    shell:
+        op.as_one_line("""
+        bcftools view -f ".,PASS" {input.vcf} -Ov
+          | 
+        bcftools sort --max-mem {resources.mem_mb}M -Oz -o {output.vcf_passed}
```

Comment:
To be safe, should we use 0.8*mem_mb here as you did in the earlier example?

Reply:
good suggestion, I have added this to the rule